base/iterators.jl

```diff
@@ -907,12 +925,17 @@ Return an iterator over the product of several iterators. Each generated element
 a tuple whose `i`th element comes from the `i`th argument iterator. The first iterator
 changes the fastest.
 
+See also: [`zip`](@ref), [`Iterators.flatten`](@ref), `Base.splat`.
```

Comment:
```suggestion
See also: [`zip`](@ref), [`Iterators.flatten`](@ref), [`Base.splat`](@ref).
```

Reply:
I think splat isn't yet in the manual. But perhaps it can be added, alongside `Base.:(|>)` and `Base.:(∘)`.

Reply:
Oh, then it shouldn't be in the "see also" list (since you can't "see" it). Seems that it got added as a helper in 43ce8999207, so was not exported

Reply:
Sure, I can remove it. 

You can't click the link on the website, but at the repl there is no obvious distinction of what is and isn't documented. Maybe whether there should be is another question. 

Note that this PR does add some other functions to docs, discussed above: https://github.com/JuliaLang/julia/pull/38606#issuecomment-765269902 . It would seem stranger for me that `promote_type` not to cross link to `promote_typejoin`, since understanding that there are two ways in which Base widens seems important for anyone looking this up.

Reply:
Right. I reviewed everything, and thought it looked good, so I'm not suggesting changing any others based on this thread.

Reply:
```suggestion
See also: [`zip`](@ref), [`Iterators.flatten`](@ref).
```